daemon/endpoint.go

```diff
@@ -159,9 +160,18 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 		return apierror.Error(PutEndpointIDFailedCode, err)
 	}
 
-	if err := ep.RegenerateIfReady(h.d); err != nil {
-		ep.RemoveDirectory()
-		return apierror.Error(PatchEndpointIDFailedCode, err)
+	// Regenerate immediately if ready
+	ep.Mutex.Lock()
+	ready := false
+	if ep.GetStateLocked() == endpoint.StateReady {
+		ready = ep.SetStateLocked(endpoint.StateWaitingToRegenerate)
+	}
+	ep.Mutex.Unlock()
+	if ready {
```

Comment:
My intention was not to change the behavior here, but upon checking this again I see that if the initial state was waiting-for-identity master would kick a build at this time, while this depended upon a following PATCH to do the initial build in that case. I've added a patch that adds a check for waiting-for-identity as well so that it behaves the same as master. Note that it is still possible that the initial state is "creating" (or "disconnected" in master) and in this case the initial build is (and was) no done at this stage, but when processing the ensuing PATCH request that changes the state to waiting-for-identity (or ready).

Reply:
I'm not 100% sure what the implication of no bpf build is between the PUT and PATCH in the case the initial state is not waiting-for-identity or ready, but that is how the master behaves, so if we want to change that, I'd suggest doing it in a separate PR.

Reply:
Makes sense! Thank you for the clear explanation.